app/src/main/java/com/example/candyshop/ui/DetailsScreenViewModel.kt

```diff
@@ -3,37 +3,69 @@ package com.example.candyshop.ui
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
-import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory.Companion.APPLICATION_KEY
-import androidx.lifecycle.viewmodel.initializer
-import androidx.lifecycle.viewmodel.viewModelFactory
-import com.example.candyshop.CandyApplication
-import com.example.candyshop.database.TempDatabase
-import com.example.candyshop.network.CandyItem
-
-class DetailsScreenViewModel(private val temporaryDb: TempDatabase) : ViewModel() {
+import androidx.lifecycle.viewModelScope
+import com.example.candyshop.api.CandyItemsRepository
+import com.example.candyshop.data.Resource
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class DetailsScreenViewModel @Inject constructor(
+    private val candyItemsRepository: CandyItemsRepository,
+    private val savedStateHandle: SavedStateHandle
+) : ViewModel() {
     var textFieldInput by mutableStateOf("")
         private set
     var showCart by mutableStateOf(false)
 
 //    private val _shoppingCartItems = mutableStateListOf<CartItem>()
 //    val shoppingCartItems: List<CartItem> = _shoppingCartItems
 
+    private val candyId = savedStateHandle.get<Int>("id")
+
+    private var _detailsState = MutableStateFlow(DetailsState())
+    val detailsState = _detailsState.asStateFlow()
+
     fun updateTextField(userInput: String) {
         textFieldInput = userInput
     }
 
-    fun getDessertById(id: Int?): CandyItem? {
-        return temporaryDb.findOne(id)
+    init {
+        getDessertById(candyId ?: -1)
     }
 
-    companion object {
-        val Factory: ViewModelProvider.Factory = viewModelFactory {
-            initializer {
-                (this[APPLICATION_KEY] as CandyApplication)
-                val temporaryDb = TempDatabase
-                DetailsScreenViewModel(temporaryDb = temporaryDb)
+    private fun getDessertById(id: Int) {
+        viewModelScope.launch {
+            _detailsState.update {
+                it.copy(isLoading = true)
+            }
+            candyItemsRepository.getCandy(id).collectLatest { result ->
```

Comment:
Im not sure if here you necessary need to use Flow stream. Maybe only suspend function will work also? Flow its good to use when we want to observe something but if its not necessary we don't have to use it